File: leetcodePractice/6:12:2020/primeFactorization.py
def primeFactorization(n):
	result = []
	up = int(math.sqrt(n))

	k = 2
	while k <= up and n > 1:
		while n % k == 0:
			n //= k
			result.append(k)
		k += 1
	
	if n > 1:
		result.append(n)
	
	return result

File: leetcodePractice/7-16-2020/primeFactorization.java
/**
 * 分解质因数 ---> O(n ^ 1/ 2)
 */
public List<Integer> primeFactorization(int n) {
    List<Integer> result = new ArrayList<>();
    int up = (int) Math.sqrt(n);

    for (int k = 2; k <= up && n > 1; ++k) {
        while (n % k == 0) {
            n /= k；
            result.add(k);
        }
    }

    if (n > 1) {
        result.add(n);
    }

    return resultl;
}